llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -2156,6 +2158,11 @@ void VPBranchOnMaskRecipe::execute(VPTransformState &State) {
   ReplaceInstWithInst(CurrentTerminator, CondBr);
 }
 
+InstructionCost VPBranchOnMaskRecipe::computeCost(ElementCount VF,
+                                                  VPCostContext &Ctx) const {
```

Comment:
Some explanation for treating this branch as free.

Reply:
Added thanks!